Lib/test/test_grammar.py

```diff
@@ -415,6 +415,27 @@ class Cbad2(C):
                 x: int
                 x.y: list = []
 
+    def test_annotations_inheritance(self):
+        # Check that annotations are not inherited by derived classes
+        class A:
+            attr: int
+        class B(A):
+            pass
+        class C(A):
+            attr: str
+        class D:
+            attr2: int
+        class E(A, D):
+            pass
+        class F(C, A):
+            pass
+        self.assertEqual(A.__annotations__, {"attr": int})
+        self.assertEqual(B.__annotations__, {})
+        self.assertEqual(C.__annotations__, {"attr" : str})
```

Comment:
Why not test D's annotations?
```suggestion
        self.assertEqual(C.__annotations__, {"attr": str})
        self.assertEqual(D.__annotations__, {"attr2": int})
```

Reply:
A test on `D` is a duplicate of the test on `A` (i.e. a class with no parent classes and a single `int` attribute) the only change is the name of the attribute. Anyway, I have added it .